src/plugins/discover/public/application/main/components/top_nav/on_save_search.tsx

```diff
@@ -167,13 +175,42 @@ const SaveSearchObjectModal: React.FC<{
   title: string;
   showCopyOnSave: boolean;
   description?: string;
-  onSave: (props: OnSaveProps & { newRowsPerPage?: number }) => void;
+  timeRestore?: boolean;
+  onSave: (props: OnSaveProps & { newTimeRestore: boolean }) => void;
   onClose: () => void;
-}> = ({ title, description, showCopyOnSave, onSave, onClose }) => {
+}> = ({ title, description, showCopyOnSave, timeRestore: savedTimeRestore, onSave, onClose }) => {
+  const [timeRestore, setTimeRestore] = useState<boolean>(savedTimeRestore || false);
+
   const onModalSave = (params: OnSaveProps) => {
-    onSave(params);
+    onSave({
+      ...params,
+      newTimeRestore: timeRestore,
+    });
   };
 
+  const options = (
+    <EuiFormRow
+      helpText={
+        <FormattedMessage
+          id="discover.topNav.saveModal.storeTimeWithSearchFormRowHelpText"
+          defaultMessage="Restore the time filter with the currently selected time range and refresh interval each time this search is opened"
+        />
+      }
+    >
+      <EuiSwitch
+        data-test-subj="storeTimeWithSearch"
+        checked={timeRestore}
+        onChange={(event) => setTimeRestore(event.target.checked)}
+        label={
+          <FormattedMessage
+            id="discover.topNav.saveModal.storeTimeWithSearchFormRowLabel"
+            defaultMessage="Store time with search"
```

Comment:
Hi @gchaps ! We are adding a new toggle to the Save Search modal on Discover page. Do you have suggestions for the copy?

<img width="580" alt="Screenshot 2022-08-09 at 13 04 02" src="https://user-images.githubusercontent.com/1415710/183733880-ff53c11d-54b5-4cf6-8ae0-98f33ff0402b.png">

Reply:
How about something like this

Store time with saved search
Update the time filter and refresh interval to the current selection when using this search.

Reply:
Updated. Thanks, @gchaps !

<img width="460" alt="Screenshot 2022-08-11 at 09 31 49" src="https://user-images.githubusercontent.com/1415710/184085727-c314db52-4389-4641-a7c2-64e940d38a5b.png">